pkg/extension/controller/utils.go

```diff
@@ -54,3 +59,55 @@ func Resolve[T any](ctx context.Context, kuadrantCtx exttypes.KuadrantCtx, polic
 	}
 	return result, nil
 }
+
+func AcceptedCondition(p exttypes.Policy, err error) *metav1.Condition {
+	policyKind := fmt.Sprintf("%s", p.GetObjectKind())
+	cond := &metav1.Condition{
+		Type:    string(gatewayapiv1alpha2.PolicyConditionAccepted),
+		Status:  metav1.ConditionTrue,
+		Reason:  string(gatewayapiv1alpha2.PolicyReasonAccepted),
+		Message: fmt.Sprintf("%s has been accepted", policyKind),
+	}
+	if err == nil {
+		return cond
+	}
+
+	policyErr := intoPolicyError(err, policyKind)
+
+	cond.Status = metav1.ConditionFalse
+	cond.Message = policyErr.Error()
+	cond.Reason = string(policyErr.Reason())
+
+	return cond
+}
+
+func EnforcedCondition(p exttypes.Policy, err error, fully bool) *metav1.Condition {
+	policyKind := fmt.Sprintf("%s", p.GetObjectKind())
+	message := fmt.Sprintf("%s has been successfully enforced", policyKind)
+	if !fully {
+		message = fmt.Sprintf("%s has been partially enforced", policyKind)
+	}
+	cond := &metav1.Condition{
+		Type:    string(kuadrant.PolicyConditionEnforced),
+		Status:  metav1.ConditionTrue,
+		Reason:  string(kuadrant.PolicyReasonEnforced),
+		Message: message,
+	}
+	if err == nil {
+		return cond
+	}
+	policyErr := intoPolicyError(err, policyKind)
+	cond.Status = metav1.ConditionFalse
+	cond.Message = err.Error()
+	cond.Reason = string(policyErr.Reason())
+
+	return cond
+}
```

Comment:
Are these defined elsewhere in kuadrant or are they coming from policy-machinery for our policies? not against this either fwiw just wondered if we could re-use something

Reply:
Cool, that sounds reasonable, I think this is good and we can refine later if needed